servers/services/src/main/java/org/projectnessie/services/hash/HashValidator.java

```diff
@@ -17,50 +17,122 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
 import javax.annotation.Nullable;
+import org.projectnessie.versioned.BranchName;
+import org.projectnessie.versioned.NamedRef;
+import org.projectnessie.versioned.TagName;
 
-@FunctionalInterface
-public interface HashValidator {
+public final class HashValidator {
```

Comment:
The refactoring is mainly motivated by the need to preserve the field names when generating error messages. Otherwise it's functionally equivalent.